service/db.hpp

```diff
@@ -80,370 +93,309 @@ namespace raven
+         *
+         * In case an error occur, the state will be set accordingly (`unknow_config_key`, `sql_error`, `fatal_error`)
+         *
+         */
+
         LOG_SCOPE_F(INFO, __PRETTY_FUNCTION__);
+        config_id_st config_id;
+        auto functor_receive_data = [&config_id](int id) { config_id = config_id_st{static_cast<std::size_t>(id)}; };
         try {
-            auto functor_receive_data = [&setting_update_data, db_id, this](const std::string json_text) {
-                auto json_data = json::json::parse(json_text);
-                for (auto&[key, value] : setting_update_data.settings_to_update.items()) {
-                    json_data["SETTINGS"][key] = value;
-                }
-                DLOG_F(INFO, "json after update: %s", json_data.dump().c_str());
-                this->execute_statement(update_config_text_from_id_statement,
-                    json_data.dump().c_str(), db_id.value());
-            };
-            throw_misuse_if_count_return_zero_for_this_statement(select_count_config_from_id_statement,
-                                                                 db_id.value());
-            execute_statement(select_config_from_id_statement, db_id.value()) >> functor_receive_data;
+            int nb_count = 0;
+            execute_statement(select_count_config_element_statement) >> nb_count;
+            if (!nb_count)
+                throw sqlite::errors::empty(0, select_count_config_element_statement.value());
+            throw_misuse_if_count_return_zero_for_this_statement(select_count_key_statement, config_key.value(), config_key.value());
+            execute_statement(select_config_from_key_statement, config_key.value(), config_key.value()) >> functor_receive_data;
+            state = db_state::good;
+        }
+        catch (const sqlite::errors::empty &error) {
+            DLOG_F(ERROR, "misuse of api or wrong key: %s, from sql: %s", error.what(), error.get_sql().c_str());
+            state = db_state::sql_error;
         }
         catch (const sqlite::errors::misuse &error) {
-            DLOG_F(ERROR, "misuse of api or wrong id: %s, from sql: %s -> [with id = %lu]", error.what(),
-                error.get_sql().c_str(),
-                db_id.value());
-            return request_state::unknown_id;
+            DLOG_F(ERROR, "misuse of api or wrong key: %s, from sql: %s", error.what(), error.get_sql().c_str());
+            state = db_state::unknow_config_key;
         }
         catch (const sqlite::sqlite_exception &error) {
             DLOG_F(ERROR, "error: %s, from sql: %s", error.what(), error.get_sql().c_str());
-            return request_state::db_error;
+            state = db_state::sql_error;
         }
         catch (const std::exception &error) {
             DLOG_F(ERROR, "%s", error.what());
-            return request_state::db_error;
+            state = db_state::fatal_error;
         }
-        return request_state::success;
+        return config_id;
```

Comment:
Does the returned `config_id` make any sens if there was an error ? Because if not this would exceptions even more logical.

Reply:
(To be clear, I don't think we should do anything in this PR. I just want to know so that I can open an issue if necessary)

Reply:
No if there was an error the return isn't viable, the idea is you have to check the state of the db after any operation to ensure the db and/or the return value are viable, but the more I think about exception and the more I want to keep going that way (and of course improve as much as possible) I think we should open an issue for that and discuss about it there
in the mean time maybe we can use std::optional here ? to make sure id doesn't have a value so that the user can't use it if he forgot to check the state of the db ?
(that also make me think I need to write a complete doc with the behavior of each function of the service)